src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts

```diff
@@ -50,18 +65,29 @@ export class TerminalShellIntegrationLinkDetector implements ITerminalLinkDetect
 		return [];
 	}
 
-	private _matches(lines: IBufferLine[]): boolean {
-		if (lines.length < linkCodes.length) {
-			return false;
+	private _matches(lines: IBufferLine[]): LinkType | undefined {
+		if (lines[0].length < activatedLink.codes.length && lines[0].length < disabledLink.codes.length) {
+			return undefined;
 		}
```

Comment:
At top level `const minLinkLength = Math.min(...links.map(e => e.length))`? Then you could do this:

```ts
if (lines[0].length < minLinkLength) {
	return undefined;
}
```